PhysicsTools/NanoAOD/python/globals_cff.py

```diff
@@ -35,3 +35,36 @@
 
 globalTables = cms.Sequence(rhoTable)
 globalTablesMC = cms.Sequence(puTable+genTable)
+
+rivetProducerHTXS = cms.EDProducer('HTXSRivetProducer',
+   HepMCCollection = cms.InputTag('myGenerator','unsmeared'),
+   LHERunInfo = cms.InputTag('externalLHEProducer'),
+   #ProductionMode = cms.string('GGF'),
+   ProductionMode = cms.string('AUTO'),
+)
+
+myMergedGenParticles = cms.EDProducer("MergedGenParticleProducer",
+     inputPruned = cms.InputTag("prunedGenParticles"),
+     inputPacked = cms.InputTag("packedGenParticles"),
+)
+myGenerator = cms.EDProducer("GenParticles2HepMCConverter",
+     genParticles = cms.InputTag("myMergedGenParticles"),
+     genEventInfo = cms.InputTag("generator"),
+     signalParticlePdgIds = cms.vint32(25), ## for the Higgs analysis
```

Comment:
The configuration differs here from the general instance of GenParticles2HepMCConverter in particlelevel_cff.py, because this one looks specifically at the Higgs vertex.
Running two instances is a waste in terms of resources, as signalParticlePdgIds is only used at the end of a long routine that does many other common things first.

Can you please test how much this contributes to the timing, to see if it's worth investing in a more efficient implementation?

Reply:
Running in 1000 Z(ll)H(bb) events, the run time per event for this module is 0.0017s/event (real). The total real processing time per event is 0.76 sec, ie the duplicate instance contributes 0.2% of the total run time.